blocks/components/editable/tinymce.js

```diff
@@ -3,7 +3,13 @@ export default class TinyMCE extends wp.element.Component {
 		this.initialize();
 	}
 
-	shouldComponentUpdate() {
+	shouldComponentUpdate( nextProps ) {
+		const isEmpty = String( nextProps.isEmpty );
+
+		if ( this.editorNode.getAttribute( 'data-is-empty' ) !== isEmpty ) {
+			this.editorNode.setAttribute( 'data-is-empty', isEmpty );
+		}
```

Comment:
If we keep it here, `componentWillReceiveProps` may fire even if this function is returning `false` and would likely be most appropriate for updating the node.

https://github.com/WordPress/gutenberg/pull/475#discussion_r115020412